src/cript/nodes/exceptions.py

```diff
@@ -57,3 +59,37 @@ def __init__(self, node_type: str, json_dict: str):
 
     def __str__(self):
         return f"JSON Serialization failed for node type {self.node_type} with JSON dict: {self.json_str}"
+
+
+class CRIPTOrphranedNodesError(CRIPTException, ABC):
+    def __init__(self, orphraned_node: "BaseNode"):
+        self.orphaned_node = orphraned_node
+
+    @abstractmethod
+    def __str__(self):
+        pass
+
+
+class CRIPTOrphranedMaterialError(CRIPTOrphranedNodesError):
+    def __init__(self, orphaned_node: "Material"):
+        super().__init__(orphaned_node)
+
+    def __str__(self):
+        ret_string = "While validating a project graph, an orphraned material node was found. "
+        ret_string += "This material is present in the graph, but not listed in the project. "
+        ret_string += "Please add the node like: `your_project.materials += [orphaned_material]`. "
+        ret_string += f"The orphraned material was {self.orphraned_node}."
+        return ret_string
+
+
+class CRIPTOrphranedExperimentError(CRIPTOrphranedNodesError):
+    def __init__(self, orphaned_node):
+        super().__init__(orphaned_node)
+
+    def __str__(self):
+        node_name = self.orphaned_node.node_type.lower()
+        ret_string = f"While validating a project graph, an orphraned {node_name} node was found. "
+        ret_string += f"This {node_name} node is present in the graph, but not listed in any experiment of the  project. "
+        ret_string += f"Please add the node like: `your_experiment.{node_name} += [orphaned_{node_name}]`. "
+        ret_string += f"The orphraned {node_name} was {self.orphraned_node}."
+        return ret_string
```

Comment:
I would change the error to

```python

class OrphranedNodeError:
    parent_node = None
    child_node = None

    def __init__(self, parent_node, child_node):
        self.parent_node = parent_node
        self.child_node = child_node

    def __str__(self) -> str:
    return (
        f"While validating the {self.parent_node.node_type} graph, an orphaned node: {self.child_node} was found. "
       f"The {self.child_node} node is present in the graph, but not listed in any {self.parent_node.node_type} "
       f"Please add the {self.child_node} to the appropriate {self.parent_node.node_type} "
       f"See documentation for more details"
    )
```

and then within the documentation we say 

```text
This error is raised when a child node is not attached to the 
appropriate parent node. For example, all material nodes used 
within a project must belong to the project inventory. If there is 
a material node that is used within a project but not a part of the 
inventory and the validation code finds it then it raises an `OrphranedNodeError`

Fixing this is simple and easy, just take the node that CRIPT Python SDK 
found a problem with and associate it with the appropriate parent via

my_project.inventory = my_orphaned_material_node
```

Reply:
Sure, we can add the parent node and documentation string.

Reply:
I don't understand how the `parent_node` dynamics should work.
The parent is always of type project, so why have it dynamic?
Unless you thought if would be possible to to that for experiment, but it is actually not possible for experiment, because we don't know which experiment of the project the orphaned nodes belong to.

Reply:
It was just something rough I was thinking of, if we can't have parent node then we could just have that hard coded and only pass in the child node. I'm happy either way.

I'm happy to brainstorm together as well and think of a good way to approach this!